backend/app/Repositories/EloquentChartVisitRepository.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Repositories;
+
+use App\Contracts\Visits\PageViewsFilterData;
+use App\DataTransformer\DataTransformerInterface;
+use App\DataTransformer\visits\ChartVisitDataTransformer;
+use App\Repositories\Contracts\ChartVisitRepository;
+use Illuminate\Support\Facades\DB;
+
+final class EloquentChartVisitRepository implements ChartVisitRepository
+{
+    public function findByFilter(PageViewsFilterData $filterData, int $interval): DataTransformerInterface
+    {
+        $result =  DB::select( (string)DB::raw("SELECT COUNT(*) as visits, date FROM (
+                    SELECT visits.*,
+                    (extract(epoch FROM created_at) - MOD( (CAST (extract(epoch FROM created_at) AS INTEGER)), :period_in_seconds)) AS date
+                   FROM visits 
+                   WHERE extract(epoch FROM created_at) >= :start_date
+                   AND
+                   extract(epoch FROM created_at) <= :end_date 
+                   ) AS periods
+                   GROUP BY date"), [
+            'period_in_seconds' => $interval,
+            'start_date' => $filterData->getStartDate()->getTimestamp(),
+            'end_date' => $filterData->getEndDate()->getTimestamp(),
+        ]);
+
+        return new ChartVisitDataTransformer($result);
```

Comment:
You do not realy need transformer here. Return raw result from repository. You can return callection, just wrapp $result in helper collection.